### PR TITLE
Add standard artefacts

### PR DESCRIPTION
**Pull request by hiker**
Replaces all strings as artefacts to enums. Users can still use strings if they need to create their own artefact sets.

### CODE REVIEW

**Comment by hiker** [comment]
Turns out I had to merge in bom_master to get git to properly detect that the changes to run_config are already there.

This PR replaces the strings defined in constant.py for artefact collections with enums in ArtefactSet - it esp. replaces the string constant `"all_source"` that is used. Otherwise it does a bit of additional cleanup. It looks big, but it is mostly change to imports (instead of constants.py the new ArtefactSet enum comes from artefact.py), and then replacing them where they are used. There is still one or two left for things that interact with e.g. the LFRic-specific configurator, but this will all properly be changed in the next PR.

File: source/fab/artefacts.py [hunk 1]
```diff
@@ -4,35 +4,109 @@
...
         super().__init__()
         self.reset()
 
     def reset(self):
         '''Clears the artefact store (but does not delete any files).
         '''
         self.clear()
-        self[CURRENT_PREBUILDS] = set()
+        for artefact in ArtefactSet:
+            if artefact == ArtefactSet.OBJECT_FILES:
```
Comment (jasonjunweilyu) [inline source/fab/artefacts.py:60]: Is `ArtefactSet.Build_TREES` also a dictionary by default? This should not matter, as `ArtefactSet.Build_TREES` is reset when used by `steps/analyse.py`.

**Review by jasonjunweilyu — CHANGES_REQUESTED** [review]
Hi @hiker , I have finished my review and thanks for adding `ArtefactSet`. There are just some small comments and then this is ready to be merged.

**Comment by hiker** [comment]
Ready for next review.

**Review by jasonjunweilyu — APPROVED** [review]
Thanks. I think now this is ready to be merged.

**Pull request merged**